packages/kit/src/core/sync/write_env.js

```diff
@@ -0,0 +1,183 @@
+import { write_if_changed } from './utils.js';
+import path from 'path';
+import fs from 'fs';
+import { loadEnv } from 'vite';
+
+const autogen_comment = '// this section is auto-generated';
+
+/**
+ * @param {boolean} pub
+ * @param {Record<string, string>} env
+ * @returns {string}
+ */
+function const_declaration_template(pub, env) {
+	return `${autogen_comment}
+
+${Object.entries(env)
+	.map(
+		([k, v]) => `/**
+ * @type {import('$app/env${pub ? '' : '/private'}').${k}}
```

Comment:
genuine question — i'm sure there's a good reason — but what's the reason to do this over just `string`?

```suggestion
 * @type {string}
```

Reply:
Generally speaking I prefer to be more rather than less explicit (especially in generated code), but tbh it probably could just be string. I just prefer the hard coupling of type definitions to where they're defined. 🤷

Reply:
I guess what I'm asking is what effect does this actually have? `Right-click -> Go To Definition` takes me to `.svelte-kit/types/ambient.d.ts` in either case